idea/idea-fir/tests/org/jetbrains/kotlin/idea/quickfix/AbstractHighLevelQuickFixTest.kt

```diff
@@ -7,20 +7,90 @@ package org.jetbrains.kotlin.idea.quickfix
 
 import com.intellij.codeInsight.intention.IntentionAction
 import com.intellij.codeInspection.InspectionProfileEntry
+import com.intellij.openapi.util.io.FileUtil
 import org.jetbrains.kotlin.test.uitls.IgnoreTests
+import java.io.File
 import java.nio.file.Paths
 
 
 abstract class AbstractHighLevelQuickFixTest : AbstractQuickFixTest() {
+    companion object {
+        private val isTeamCityBuild: Boolean = System.getProperty("TEAMCITY_VERSION") != null
+    }
+
     override fun doTest(beforeFileName: String) {
         IgnoreTests.runTestIfEnabledByFileDirective(
             Paths.get(beforeFileName),
-            enableTestDirective = IgnoreTests.DIRECTIVES.FIR_COMPARISON_MUTLTILINE_COMMENT,
+            enableTestDirective = IgnoreTests.DIRECTIVES.FIR_COMPARISON_MULTILINE_COMMENT,
             directivePosition = IgnoreTests.DirectivePosition.LAST_LINE_IN_FILE,
             additionalFilesExtensions = arrayOf("after")
         ) {
             super.doTest(beforeFileName)
+            // getFir/OriginalAfterFile expect just the filename (no directory), but beforeFileName has the directory
+            val beforeFileNameOnly = File(beforeFileName).name
+            val firAfterFile = getFirAfterFile(beforeFileNameOnly)
+            val originalAfterFile = getOriginalAfterFile(beforeFileNameOnly)
+            if (firAfterFile.exists() && firAfterFile.readText().trim() == originalAfterFile.readText().trim()) {
+                // The FIR after file is identical with the original after file. Remove the FIR after file and mark "FIR_IDENTICAL" in the original files.
+                // Note: FIR_IDENTICAL cannot be at the start of the file because the test uses ActionHint which parses the first line to get
+                // the action to perform (or check that they are not available).
+                firAfterFile.delete()
+
+                fun appendFirIdentical(file: File) {
+                    // The file ends with FIR_COMPARISON directive. It has to remain the last line for the test to be enabled, FIR_IDENTICAL must go before that.
+                    val lines = file.readLines().dropLast(1)
+                    file.writer().use {
+                        for (line in lines) {
+                            it.appendLine(line)
+                        }
+                        it.appendLine(IgnoreTests.DIRECTIVES.FIR_IDENTICAL)
+                        it.appendLine(IgnoreTests.DIRECTIVES.FIR_COMPARISON_MULTILINE_COMMENT)
+                    }
+                }
+
+                val beforeFile = File(beforeFileName)
+                appendFirIdentical(beforeFile)
```

Comment:
Do we really need that additional FIR_IDENCITCAL directive? we already have FIR_COMPARISON. 
If we have a .fir.after file, we already know that we should use it as the actual testdata

Reply:
We could go either way. We use FIR_IDENTICAL in many of the FIR compiler tests as a way to be _explicit_ about the test expectation that it is identical for FIR. We could say the same about those compiler tests...

Let me know if you prefer that I remove it, I'll do so.

Reply:
I just went ahead and removed the FIR_IDENTICAL directive

Reply:
I just do not see a benefit we get by adding FIR_IDENTICAL directive. For FIR compiler this is probably important because compiler want to preserve FE1.0 behaviour, while FIR IDE do not have such goal. So, I'm for removing